src/d3a/d3a_core/export.py

```diff
@@ -113,10 +125,12 @@ def export(self):
         self.plot_all_unmatched_loads()
         self.plot_avg_trade_price(self.area, self.plot_dir)
         self.plot_ess_soc_history(self.area, self.plot_dir)
-        self.move_root_plot_folder()
+        if ConstSettings.GeneralSettings.EXPORT_DEVICE_PLOTS:
+            self.plot_device_stats(self.area, [])
         if ConstSettings.IAASettings.MARKET_TYPE == 3 and \
                 ConstSettings.GeneralSettings.SUPPLY_DEMAND_PLOTS:
             self.plot_supply_demand_curve(self.area, self.plot_dir)
+        self.move_root_plot_folder()
```

Comment:
@faizan2590 FYI: This change (moving `self.move_root_plot_folder()` further down fixes an issue with the supply-demand-plots. If you like, i can explain offline.